lib/backend/syncersv1/updateprocessors/rules.go

```diff
@@ -40,6 +40,45 @@ func RulesAPIV2ToBackend(ars []apiv3.Rule, ns string) []model.Rule {
 	return brs
 }
 
+// entityRuleAPIV2ToBackend collects the ordered set of selectors for the EntityRule:
+// (serviceAccountSelector) && (selector)
+// It also returns the namespace selector to use
+func entityRuleAPIV2TOBackend(er *apiv3.EntityRule, ns string) (nsSelector, selector string) {
+
+	// Determine which namespaces are impacted by this entityRule.
+	if er.NamespaceSelector != "" {
+		// A namespace selector was given - the rule applies to all namespaces
+		// which match this selector.
+		nsSelector = parseSelectorAttachPrefix(er.NamespaceSelector, conversion.NamespaceLabelPrefix)
+	} else if ns != "" {
+		// No namespace selector was given and this is a namespaced network policy,
+		// so the rule applies only to its own namespace.
+		nsSelector = fmt.Sprintf("%s == '%s'", apiv3.LabelNamespace, ns)
+	}
+
+	var selectors []string
+
+	// Determine which service account selector.
+	if er.ServiceAccounts != nil {
```

Comment:
@caseydavenport because we check for the flag at lib/clientv3/[gnp|np] layer. it is safe to assume that when we come to this point that the ServiceAccounts match would be nil if the alphaSA flag was not set.

Reply:
I think this is still an issue because the client code runs as a totally separate process from this.

So, someone could enable the feature on the client, disable it in Felix, and not expect Felix to interpret it.

Reply:
k. i see your point. let me see if i can get the flag all the way down :-)

Reply:
@caseydavenport It is going to be a bit hairy getting the `AlphaFeatures` peculating all the way down here. given the tight timing and the remaining review, can we -for now- assume that the client is the gatekeeper? I can add this as an enhancement after this pr is merged. wdyt?

Reply:
SGTM.

As discussed, we can also just mandate an env var be set, and use `os.Getenv` to avoid having to pass the flag around to so many places.